src/main/java/org/opensearch/geospatial/ip2geo/dao/Ip2GeoCachedDao.java

```diff
@@ -37,52 +40,70 @@
 @Log4j2
 public class Ip2GeoCachedDao implements IndexingOperationListener {
     private final DatasourceDao datasourceDao;
-    private Map<String, DatasourceMetadata> data;
+    private final GeoIpDataDao geoIpDataDao;
+    private final GeoDataCache geoDataCache;
+    private Map<String, DatasourceMetadata> metadata;
 
-    public Ip2GeoCachedDao(final DatasourceDao datasourceDao) {
+    public Ip2GeoCachedDao(final ClusterService clusterService, final DatasourceDao datasourceDao, final GeoIpDataDao geoIpDataDao) {
         this.datasourceDao = datasourceDao;
+        this.geoIpDataDao = geoIpDataDao;
+        this.geoDataCache = new GeoDataCache(clusterService.getClusterSettings().get(Ip2GeoSettings.CACHE_SIZE));
```

Comment:
How do we make sure there is only one instance of GeoDataCache?

Reply:
maybe add a code comment explaining this reasoning